### DISCUSSION TITLE
post/powerbi-odata-dynamic-table/

### DISCUSSION THREAD

**Discussion by giscus** [AlexanderHenkel/blogComments#10] [Announcements]
# post/powerbi-odata-dynamic-table/

Learn how to query OData with dynamics dates to avoid one or more tables references a dynamic data source error

https://alexanderhenkel.dk/post/powerbi-odata-dynamic-table/

**Comment by pyepez** [comment]
Thanks for this helpful post. It works great, but I have to connect using windows authentication throught a gateway. In this scenario, I get a credential error in the scheduled refresh, what else need to be done in this case?

**Reply by AlexanderHenkel** [reply]
If you are running trough a **data gateway** the credentials should be created in **manage connections and gateways**

**Comment by hexekutive** [comment]
This is incredible!!! Thank you so much for writing this post. A year ago, a tried and tried different methods, but none worked. Now, I found your blog and it solved the problem!!! THANKS A LOT!!!

**Reply by AlexanderHenkel** [reply]
So glad to hear that it helped you. I did also spend way to much time finding this solution, but it has helped me in many projects since
